tests/ballerina-tools-integration-test/src/test/java/org/ballerinalang/test/packaging/ModuleBuildTestCase.java

```diff
@@ -211,6 +211,78 @@ public void testBuildEmptyProject() throws BallerinaTestException, IOException {
         clientLeecher.waitForText(3000);
     }
 
+    /**
+     * Building a project with a module only with tests.
+     *
+     * @throws BallerinaTestException When an error occurs executing the command.
+     */
+    @Test(description = "Test building a project with a module only with tests")
+    public void testBuildOnlyWithTest() throws BallerinaTestException, IOException {
+        Path projectPath = tempProjectDirectory.resolve("testModuleBuild");
+        initProject(projectPath, SINGLE_PKG_PROJECT_OPTS);
+
+        // Delete the source files in the module
+        Files.deleteIfExists(projectPath.resolve("foo").resolve("main.bal"));
```

Comment:
This statement is written based on an assumption, that project directory has only main.bal. If someone adds foo.bal, this test case still passes, but it doesn't test the actual scenario. Ideally, we should delete all files and folders inside the package, except the test folder and its files.

Reply:
Added another method to create a project with only test sources with cc7c6c8

Reply:
Refactored the test case with 390c06e